src/NServiceBus.Core/Class1.cs

```diff
@@ -0,0 +1,21 @@
+﻿using NServiceBus.Extensibility;
+
+/// <summary>
+/// test.
+/// </summary>
+public static class Extensions
+{
+    /// <summary>
+    /// test.
+    /// </summary>
+    public static ContextBag GetRootContext(this ContextBag incomingContext)
+    {
+        ContextBag current = incomingContext;
```

Comment:
Discussed with Laila and another approach to consider is add an protected `rootContext` field that could be auto-populated in the ctor based on the parentContext so we don't have to traverse the hierarchy every time.